### ISSUE TITLE
Enable/disable drawer via API call

### ISSUE THREAD

**Issue by pre** [AppGyver/steroids#749]

**Comment by modess** [comment]
+1

**Comment by bFlood** [comment]
+1
we need a way to enable the right drawer on only certain webviews. disabling gestures works for disabling the drawers but it disables both drawers at once

**Comment by pentateu** [comment]
In Android the Drawer must be added to the view stack at the start of the application, so creating the Drawer in runtime when calling the API causes ugly view refresh/repaints on slower devices.

We have an API to configure the gestures that can be used to disable the drawer, but that affects both sides.

So I propose the addition of a simple API to enable/disable a specific side of the drawer:
```
@testDisableLeftDrawer: ->
    steroids.drawers.disable
      side: "left"
    ,
      onSuccess: -> steroids.logger.log "SUCCESS left drawer disabled"
      onFailure: -> navigator.notification.alert "FAILURE in testDisableLeftDrawer"
      
  @testEnableLeftDrawer: ->
    steroids.drawers.enable
      side: "left"
    ,
      onSuccess: -> steroids.logger.log "SUCCESS left drawer enabled"
      onFailure: -> navigator.notification.alert "FAILURE in testEnableLeftDrawer"
```

**Comment by andybbtt11** [comment]
So is there no way to update the width of the drawer after runtime on Android? eg make a viewport-width drawer?

**Comment by harsaharsa** [comment]
This is available in Android 4.0.6 and iOS 4.0.4-rc1. Will close the issue once 4.0.4 is out for good.

**Comment by harsaharsa** [comment]
@andybbtt11 the drawer width should be changeable with `supersonic.ui.drawers.update`, do you have a piece of code that fails?

**Comment by andybbtt11** [comment]
@harsaharsa I have tried supersonic.ui.drawers.updateOptions with width, and widthOfDrawerInPixels. It seems as if the updateOptions method doesn't accept any type of width arguments. I am able to update the width of the drawer in iOS using supersonic.ui.drawers.init( iosDrawer, options ), but android doesn't support the init method. And supersonic.ui.drawers.update() is undefined. Thanks.

**Comment by harsaharsa** [comment]
@andybbtt11 you're right, there's a bug/missing feature in Supersonic – the `supersonic.ui.drawers.init` is actually calling `steroids.drawers.update` internally, but just returns if on Android, and the `updateOptions` doesn't pass any width arguments.

We'll figure out a way to expose the correct APIs to Supersonic, but in the meantime, you can call the Steroids API directly (example in CoffeeScript):

```
steroids.drawers.update {
  left:
    widthOfDrawerInPixels: 200
}, {
  onSuccess: successFunction
  onFailure: failureFunction
}
```
You can also pass in the `right` object to change right drawer width; or change the width of both drawers by specifying the width of the visible layer (0 is not supported though):
```
steroids.drawers.update {
  options:
    widthOfLayerInPixels: 20
}
```

Sorry for the confusion!

**Comment by andybbtt11** [comment]
I got it to work, thanks a ton @harsaharsa !!

**Comment by satusuuronen** [comment]
Available in iOS 4.0.4.
